lib/upperkut/batch_execution.rb

```diff
@@ -11,31 +11,26 @@ def initialize(worker, logger = Upperkut::Logging.logger)
 
     def execute
       worker_instance = @worker.new
-      items = @worker.fetch_items.freeze
-
-      items_body = items.collect do |item|
-        item['body']
-      end
+      items = @worker.fetch_items
 
       @worker.server_middlewares.invoke(@worker, items) do
-        worker_instance.perform(items_body.dup)
+        worker_instance.perform(items)
```

Comment:
we still need to dup because #perform should receive the body only and can mutate its contents.